libminifi/src/controllers/keyvalue/PersistableKeyValueStoreService.cpp

```diff
@@ -29,20 +29,31 @@ PersistableKeyValueStoreService::PersistableKeyValueStoreService(const std::stri
 
 PersistableKeyValueStoreService::~PersistableKeyValueStoreService() = default;
 
-bool PersistableKeyValueStoreService::setImpl(const std::string& key, const std::string& value) {
-  return set(key, value);
+bool PersistableKeyValueStoreService::setImpl(const utils::Identifier& key, const std::string& value) {
+  return set(key.to_string(), value);
 }
 
-bool PersistableKeyValueStoreService::getImpl(const std::string& key, std::string& value) {
-  return get(key, value);
+bool PersistableKeyValueStoreService::getImpl(const utils::Identifier& key, std::string& value) {
+  return get(key.to_string(), value);
 }
 
-bool PersistableKeyValueStoreService::getImpl(std::unordered_map<std::string, std::string>& kvs) {
-  return get(kvs);
+bool PersistableKeyValueStoreService::getImpl(std::map<utils::Identifier, std::string>& kvs) {
+  std::unordered_map<std::string, std::string> states;
+  if (!get(states)) {
+    return false;
+  }
+  kvs.clear();
+  for (const auto& state : states) {
+    utils::optional<utils::Identifier> optional_uuid = utils::Identifier::parse(state.first);
+    if (optional_uuid) {
+      kvs[optional_uuid.value()] = state.second;
+    }
```

Comment:
since a `PerisistableKeyValueStoreService (PKVSS)` ISA `KeyValueStoreService (KVSS)` (publicly inherits) which can store arbitrary `std::string` keys, it is not impossible to cast a `PKVSS` to a `KVSS`, add some "invalid" key, and "mess up" the `PKVSS`, I would refactor this to not inherit from `KVSS` but contain an instance of it, for now I will add a log message

Reply:
Sounds like a violation of the Liskov substitution principle. Thanks for the mitigation.